src/environment.js

```diff
@@ -954,6 +954,39 @@ export class LexicalEnvironment {
   parent: null | LexicalEnvironment;
   realm: Realm;
 
+  partiallyEvaluateCompletionDeref(
+    ast: BabelNode, strictCode: boolean, metadata?: any
+  ): [Completion | Value, BabelNode, Array<BabelNodeStatement>] {
+    let [result, partial_ast, partial_io] = this.partiallyEvaluateCompletion(ast, strictCode, metadata);
+    if (result instanceof Reference) {
+      result = GetValue(this.realm, result);
+    }
+    return [result, partial_ast, partial_io];
+  }
+
+  partiallyEvaluateCompletion(
+    ast: BabelNode, strictCode: boolean, metadata?: any
```

Comment:
I feel like a comment explaining the third argument on one of these functions would be useful, as its not intuitive to me that i/o includes modifications to global state. Or is this the common phrase in the literature?